Projects/Server/Utilities/RefPool.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Server.Utilities
+{
+  public interface IRef : IDisposable { }
+
+  public class RefPool<TRef> where TRef : IRef
+  {
+    private ConcurrentBag<TRef> m_Queues = new ConcurrentBag<TRef>();

```

Comment:
Do we need a concurrent bag? I'm debating this internally since everything will be on the game thread (single threaded) and concurrent bag uses locks.
Maybe a concurrent queue since it doesn't use locks? Or a regular List Since we may not need concurrency?

Reply:
Sorry, I tend to work in a lot concurrent environments, If ModernUO is not planning on multi-threaded execution, I'm totally fine with changing it to a simple collection type. Maybe a stack or queue? I think they both use a linked-list like structure that are both fast for add/remove.